lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -1017,6 +1017,11 @@ private static final String hmac(String text, String key) {
 	 * @throws AblyException
 	 */
 	public void setClientId(String clientId) throws AblyException {
+		if(clientId == null) {
+			/* do nothing - we received a token without a clientId */
```

Comment:
If this is possible, shouldn't it be in the spec and thus have an associated test?